crates/sui-gateway/src/event_api.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::BTreeMap;
+use std::sync::Arc;
+
+use jsonrpsee_core::error::SubscriptionClosed;
+use jsonrpsee_core::server::rpc_module::PendingSubscription;
+use jsonrpsee_proc_macros::rpc;
+use serde::Deserialize;
+use serde::Serialize;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::Sender;
+use tokio_stream::wrappers::BroadcastStream;
+use tokio_stream::StreamExt;
+use tracing::{debug, warn};
+
+use sui_core::gateway_types::SuiEvent;
+
+#[rpc(server, client, namespace = "sui")]
+pub trait EventApi {
+    #[subscription(name = "subEvent", unsubscribe = "unsubEvent", item = SuiEvent)]
+    fn sub_event(&self, key: EventType);
+}
+
+#[derive(Ord, PartialOrd, Eq, PartialEq, Serialize, Deserialize, Copy, Clone, Debug, EnumIter)]
+pub enum EventType {
+    Foo,
+    Bar,
+}
+
+pub struct SuiEventManager {
+    txs: BTreeMap<EventType, Sender<SuiEvent>>,
```

Comment:
does `txn` mean `transactions`? Is `subscription` a better name?

Reply:
tx means transmit here, I will rename it